### ISSUE TITLE
[TF 2.0] Logs from model.fit is not conform to the documentation for Molde with multiple outputs

### ISSUE THREAD

**Issue by jrabary** [tensorflow/tensorflow#26799]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- TensorFlow version (use command below): tf-nightly-gpu-2.0-preview

**Describe the current behavior**

I have a model with multiple outputs. Precisely, an object detector which outputs locations and classes of object. The model is compiled as follow:

```
ssd.compile(optimizer=optimizer, 
            loss={'output_1': loc_loss, 'output_2': class_loss},
            loss_weights=[1., 1.]
           )
```

From the documentation, the loss to be optimized should be `loss = loc_loss + class_loss` but
fit outputs the following logs to the console:

```
Epoch 1/5
  37/1000 [>.............................] - ETA: 23:46 - loss: 176724.1514 - output_1_loss: 0.0212 - output_2_loss: 0.0449
```

Clearly, the value of `loss` is not the sum of the two losses.

Is it a real bug or a misunderstanding on the meaning of `output_1_loss` and `output_2_loss` ?

**Comment by jrabary** [comment]
After some investigation, I think the problem is that I don't use the same reduction in my loss ( a custom loss with Reduction.SUM) and the loss metrics which use `SumOverBatchSize` :

```
    if len(self.outputs) > 1:
      self._output_loss_metrics = [
          metrics_module.SumOverBatchSize() if hasattr(loss_fn, 'reduction')
          else metrics_module.SumOverBatchSizeMetricWrapper(loss_fn)
          for loss_fn in self.loss_functions
      ]
```

Does it make sense to have the ability to set a custom metric here ?

**Comment by jvishnuvardhan** [comment]
@jrabary I am closing the issue as it was resolved. Check this [resource](https://github.com/tensorflow/tensorflow/issues/30711) for custom_metric implementation. Thanks!
